torch/csrc/api/include/torch/nn/options/activation.h

```diff
@@ -56,5 +57,20 @@ struct LeakyReLUOptions {
   TORCH_ARG(bool, inplace) = false;
 };
 
+// ============================================================================
+
+/// Options for the Softmax functional and module.
+struct TORCH_API SoftmaxOptions {
+  SoftmaxOptions(int dim = -1, torch::Dtype dtype = torch::Dtype::Undefined);
+
+  // Dimension along which Softmax will be computed.
+  TORCH_ARG(int, dim);
+
+  // The desired data type of returned tensor.
+  // If specified, the input tensor is casted to dtype before the operation
+  // is performed. This is useful for preventing data type overflows.
+  TORCH_ARG(torch::Dtype, dtype);
```

Comment:
I think semantically `dtype` is not one of the options of a softmax function/module, because it doesn't control how the softmax function itself should perform the computation, but rather how the returned tensor should be stored, which is rather mechanical than mathematical.

Reply:
OK, deleted `dtype` from `SoftmaxOptions`.